dist/src/main/java/io/camunda/application/commons/identity/AuthenticationConfiguration.java

```diff
@@ -7,11 +7,13 @@
  */
 package io.camunda.application.commons.identity;
 
+import io.camunda.zeebe.gateway.rest.ConditionalOnRestGatewayEnabled;
 import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Profile;
 
 @ComponentScan(basePackages = {"io.camunda.authentication"})
 @ConfigurationPropertiesScan(basePackages = {"io.camunda.authentication"})
 @Profile("auth-basic")
+@ConditionalOnRestGatewayEnabled
```

Comment:
❓ I think this means that the auth layer will only be included when the `Rest` controllers are enabled which I think is incorrect, the auth layer should be enabled for all controllers and not just the rest gateway enabled ones

Reply:
It's because `CamundaUserDetailsService` requires UsersServices and the bean for the service is only available when `@ConditionalOnRestGatewayEnabled` passed (See` CamundaServicesConfiguration`). should we change this configuration? any thoughts here?

Reply:
Sorry @maryarm I was misreading this annotation, I ready it as `@ConditionalOnRestQueryEnabled` 🤦🏻 - this is fine as it is correct annotated